modules/opentelemetry/src/main/scala/zio/opentelemetry/tracing/SpanUtils.scala

```diff
@@ -0,0 +1,22 @@
+package zio.opentelemetry.tracing
+
+import io.opentelemetry.trace._
+import zio.clock.Clock
+import zio.{ Cause, UIO, URIO }
+
+private[opentelemetry] object SpanUtils {
+
+  /**
+   * Sets the status of `span` to `UNKNOWN` error with description being the pretty-printed cause.
+   */
+  def setErrorStatus(span: Span, cause: Cause[_]): UIO[Unit] =
+    UIO(span.setStatus(Status.UNKNOWN.withDescription(cause.prettyPrint)))
+
+  /**
+   * Ends the span with the current time.
+   */
+  def endSpan(span: Span): URIO[Clock, Unit] = {
```

Comment:
Would we want to have this function inside `Tracing.Service`? Right now, the consumers of this library would need to provide `Clock` twice (as shown in the example):
```scala
Clock.live ++ (Clock.live >>> Tracing.live(tracer))
```
1 time for determining starting timestamps, another for ending timestamps. By having `endSpan` in `Tracing.Service`, that would be remedied.

Reply:
Absolutely, let's do this. Changing opentracing as you propose has been on the TODO list for a few weeks, but we never got around to doing it. `Clock` should only be needed when constructing the layer. We can adjust opentracing analogously in a separate PR.

Reply:
Fixed with https://github.com/zio/zio-telemetry/pull/119/commits/2537dae04e0fb851317f3c076ff555bc526a501a